### PR TITLE
Correctly compute key range end in eetcd_kv:with_prefix/1

### LINKED ISSUE #14: eetcd_kv:with_prefix/1 is broken
`eetcd_kv:with_prefix/1` is supposed to return all the keys that start with a certain prefix, instead it returns all the keys that are greater than or equal to that prefix.

I believe the bug is here: https://github.com/zhongwencool/eetcd/blob/master/src/eetcd_kv.erl#L50. The correct behavior is described in the proto file comments here: https://github.com/etcd-io/etcd/blob/master/etcdserver/etcdserverpb/rpc.proto#L410.

### PR DESCRIPTION
**Pull request by michaelklishin**
Closes #14 which follows the idea mentioned in https://github.com/etcd-io/etcd/blob/master/etcdserver/etcdserverpb/rpc.proto#L410.

This includes #17, so #17 should be merged so that Git history makes a bit more sense.

### CODE REVIEW

**Comment by michaelklishin** [comment]
I now see what case this does not handle but `etcdctl`'s implementation does. I will look into handling that case tomorrow.

**Comment by michaelklishin** [comment]
OK, why wait if the condition is known and there's `lists:splitwith/2` :D LGTM now and still passes all tests. This is similar to what @zhongwencool had in mind in #14 but the split condition is based on characters being greater than `0xff` (and thus "unincrementable" without overflowing an unsigned char).

File: src/eetcd_kv.erl [hunk 1]
```diff
@@ -366,3 +368,29 @@ txn(Context, If, Then, Else) ->
...
+
+%%
+%% Implementation
+%%
+
+-define(UNBOUND_RANGE_END, "\0").
+
+get_prefix_range_end(Key) ->
+    case eetcd_data_coercion:to_list(Key) of
+        [] -> ?UNBOUND_RANGE_END;
```
Comment (michaelklishin) [inline src/eetcd_kv.erl]: Using a blank key as a prefix makes no sense so I'm not sure what we should do here. Returning all keys makes most sense to me.

**Comment by zhongwencool** [comment]
Thanks! ❤️ ❤️ ❤️

**Pull request merged**